src/main/java/uk/gov/pay/api/exception/mapper/CreateAgreementExceptionMapper.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.pay.api.exception.mapper;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.api.exception.CreateAgreementException;
+import uk.gov.pay.api.model.PaymentError;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+import static uk.gov.pay.api.model.PaymentError.Code.CREATE_PAYMENT_ACCOUNT_ERROR;
+import static uk.gov.pay.api.model.PaymentError.Code.CREATE_PAYMENT_CONNECTOR_ERROR;
+import static uk.gov.pay.api.model.PaymentError.aPaymentError;
+
+public class CreateAgreementExceptionMapper implements ExceptionMapper<CreateAgreementException> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CreateAgreementExceptionMapper.class);
+
+    @Override
+    public Response toResponse(CreateAgreementException exception) {
+
+        PaymentError paymentError;
+
+        if (exception.getErrorStatus() == NOT_FOUND.getStatusCode()) {
```

Comment:
this might look a bit better with a ternary operator ie.
``` 
PaymentError paymentError = exception.getErrorStatus() == NOT_FOUND.getStatusCode() ? 
aPaymentError(CREATE_PAYMENT_ACCOUNT_ERROR) : aPaymentError(CREATE_PAYMENT_CONNECTOR_ERROR)
```